app/scenes/Document/components/EditableTitle.tsx

```diff
@@ -43,175 +48,223 @@ type Props = {
+            slice = paste.slice(0);
           }
+        } else {
+          const defaultSlice = __parseFromClipboard(
+            view,
+            text,
+            html,
+            false,
+            view.state.selection.$from
+          );
+
+          // remove first node from slice
+          slice = defaultSlice.content.firstChild
+            ? new Slice(
+                defaultSlice.content.cut(
+                  defaultSlice.content.firstChild.nodeSize
+                ),
+                defaultSlice.openStart,
+                defaultSlice.openEnd
+              )
+            : defaultSlice;
         }
-      },
-      [editor]
-    );
-
-    const emojiWidth = useEmojiWidth(document.emoji, {
-      fontSize,
-      lineHeight,
-    });
-
-    const value =
-      !document.title && readOnly ? document.titleWithDefault : document.title;
-
-    return (
-      <Title
-        onClick={handleClick}
-        onChange={handleChange}
-        onKeyDown={handleKeyDown}
-        onPaste={handlePaste}
-        onBlur={onBlur}
-        placeholder={placeholder}
-        value={value}
-        $emojiWidth={emojiWidth}
-        $isStarred={document.isStarred}
-        autoFocus={!document.title}
-        maxLength={DocumentValidation.maxTitleLength}
-        readOnly={readOnly}
-        dir="auto"
-        ref={ref}
-      >
-        {starrable !== false && <StarButton document={document} size={32} />}
-      </Title>
-    );
-  }
-);
+
+        if (slice) {
+          view.dispatch(
+            view.state.tr
+              .setSelection(Selection.atStart(view.state.doc))
+              .replaceSelection(slice)
+          );
+        }
+      }
+    },
+    [editor]
+  );
+
+  const handleEmojiChange = React.useCallback(
+    async (emoji: string | null) => {
+      // Restore focus on title
+      ref.current?.focus();
+      if (document.emoji !== emoji) {
+        document.emoji = emoji;
+        await document.save();
+      }
+    },
+    [document, ref]
+  );
+
+  const handleFocus = React.useCallback(() => {
+    setFocus(true);
+  }, [setFocus]);
+
+  const value =
+    !document.title && readOnly ? document.titleWithDefault : document.title;
+
+  return (
+    <Title
+      onClick={handleClick}
+      onChange={handleChange}
+      onKeyDown={handleKeyDown}
+      onPaste={handlePaste}
+      onFocus={handleFocus}
+      onBlur={handleBlur}
+      placeholder={placeholder}
+      value={value}
+      $isStarred={document.isStarred}
+      $isFocused={isFocused}
+      $containsEmoji={!!document.emoji}
+      autoFocus={!document.title}
+      maxLength={DocumentValidation.maxTitleLength}
+      readOnly={readOnly}
+      dir="auto"
+      ref={ref}
+    >
+      {can.update ? (
+        <EmojiPicker
+          value={document.emoji}
+          onChange={handleEmojiChange}
+          // Restore focus on title
+          onClickOutside={handleClick}
+        />
+      ) : document.emoji ? (
+        <EmojiButton size={32}>
```

Comment:
A bit weird to use a component called `EmojiButton` for the non-editable state?

Reply:
Hmm, something like `<EmojiWrapper>`, perhaps?